tidy3d/plugins/smatrix/ports/coaxial_lumped.py

```diff
@@ -0,0 +1,379 @@
+"""Lumped port specialization with an annuluar geometry for exciting coaxial ports."""
+
+import pydantic.v1 as pd
+import numpy as np
+
+from ....constants import MICROMETER
+from ....components.geometry.base import Box
+from ....components.geometry.utils_2d import increment_float
+from ....components.types import FreqArray, Axis, Coordinate, Direction
+from ....components.base import cached_property
+from ....components.grid.grid import Grid, YeeGrid
+from ....components.lumped_element import CoaxialLumpedResistor
+from ....components.monitor import FieldMonitor
+from ....components.data.sim_data import SimulationData
+from ....components.source import CustomCurrentSource, GaussianPulse
+from ....components.validators import skip_if_fields_missing
+from ....components.data.data_array import FreqDataArray, ScalarFieldDataArray
+from ....components.data.dataset import FieldDataset
+from ....exceptions import SetupError, ValidationError
+
+from ...microwave import VoltageIntegralAxisAligned, CustomCurrentIntegral2D
+
+from .base_lumped import AbstractLumpedPort
+
+
+class CoaxialLumpedPort(AbstractLumpedPort):
+    """Class representing a single coaxial lumped port
+
+    Example
+    -------
+    >>> port1 = CoaxialLumpedPort(center=(0, 0, 0),
+    ...             outer_diameter=4,
+    ...             inner_diameter=1,
+    ...             normal_axis=2,
+    ...             direction="+",
+    ...             name="coax_port_1",
+    ...             impedance=50
+    ...         )
+    """
+
+    center: Coordinate = pd.Field(
+        (0.0, 0.0, 0.0),
+        title="Center",
+        description="Center of object in x, y, and z.",
+        units=MICROMETER,
+    )
+
+    outer_diameter: pd.PositiveFloat = pd.Field(
+        ...,
+        title="Outer Diameter",
+        description="Diameter of the outer coaxial circle.",
+        units=MICROMETER,
+    )
+
+    inner_diameter: pd.PositiveFloat = pd.Field(
+        ...,
+        title="Inner Diameter",
+        description="Diameter of the inner coaxial circle.",
+        units=MICROMETER,
+    )
+
+    normal_axis: Axis = pd.Field(
+        ...,
+        title="Normal Axis",
+        description="Specifies the axis which is normal to the concentric circles.",
+    )
+
+    direction: Direction = pd.Field(
+        ...,
+        title="Direction",
+        description="Specifies which side of the port the inner conductor is extending from. "
```

Comment:
It is sort of equivalent, but not exactly. The direction parameter has no effect on the fields produced by the underlying current source. LumpedPorts cannot provide that type of directional behavior, since they only contain electrical currents. 

This direction parameter is needed solely for positioning the current path integral, which needs to be placed in such a way that it surrounds a conduction current, which means it needs to surround some type of conductor or else the displacement current will make the computation less accurate. 

For the coaxial port the best method I could come up with is using the inner conductor that should be present in the `Simulation`. However, I need the user to tell me which side of the port the inner conductor is positioned. I am open to suggestions though.

Reply:
Probably clearer to say "the direction of the signal travelling in the transmission line"? Somteimes I found it hard to visualize "the side of the port the inner conductor is extending from", because I usually picture that the conductor is starts from the port.